lib/idempotency.rb

```diff
@@ -42,6 +50,7 @@ def use_cache(request, request_identifiers, lock_duration:) # rubocop:disable Me
 
     if (cached_status, cached_headers, cached_body = cached_response)
       cached_headers.merge!(Constants::HEADER_KEY => idempotency_key)
+      metric_name = 'idempotency_cache_hit_count'
```

Comment:
Could we setup a callback/hook here to call instead? Then we don't have to mix the monitoring logic & the caching logic.